src/core/Makefile.am

```diff
@@ -31,13 +31,18 @@ include $(abs_top_nlbuild_autotools_dir)/automake/pre.am
 lib_LIBRARIES                       = \
     libopenthread-ftd.a               \
     libopenthread-mtd.a               \
+    libopenthread-radio.a             \
     $(NULL)
 
 CPPFLAGS_COMMON                     = \
     -I$(top_srcdir)/include           \
     $(OPENTHREAD_TARGET_DEFINES)      \
     $(NULL)
 
+libopenthread_radio_a_CPPFLAGS      = \
+    $(CPPFLAGS_COMMON)                \
+    -DOPENTHREAD_RADIO=1              \
```

Comment:
Basically we have two categories about the API provided through UART: cli and NCP. Then about the thread capabilities : mtd and ftd. I think the mode this PR added is more about capability than API. Potentially cli can implement this mode too. So I incline to name it similar to mtd and ftd. If we have to choose between the two you give, I prefer the latter one. Thoughts? @jwhui